lib/diagnostic.js

```diff
@@ -6,23 +6,44 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  let words = paragraph.split(/\s+/);
+  let normalizedWords = [];
+  for (let i = 0; i < words.length; i++) {
+    let word = words[i].replace(/\W+/, '');
+    normalizedWords[i] = word.toUpperCase();
+  }
+
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (paragraph) {
+  let normalizedWords = normalizeWords(paragraph);
+  let uniqueWords = {};
+  for (let i = 0; i < normalizedWords.length; i++) {
+    let word = normalizedWords[i];
+    uniqueWords[word] = true;
+  }
+
+  return uniqueWords;
```

Comment:
This function returns an object with the words as keys, and `true` as the value, but the prompt specified that it should return an array.